src/main/java/com/direwolf20/buildinggadgets/common/tools/ToolRenders.java

```diff
@@ -627,49 +472,113 @@ private static void renderBox(Tessellator tessellator, BufferBuilder bufferBuild
+
+        private static IBlockState getStartBlock(RayTraceResult lookingAt, EntityPlayer player) {
+            IBlockState startBlock = Blocks.AIR.getDefaultState();
+            if (lookingAt != null)
+                startBlock = player.world.getBlockState(lookingAt.getBlockPos());
+
+            return startBlock;
+        }
+
+        private static int getStackEnergy(ItemStack stack, EntityPlayer player) {
+            if (player.capabilities.isCreativeMode || (!stack.hasCapability(CapabilityEnergy.ENERGY, null) && !stack.isItemStackDamageable()))
+                return 1000000;
+
+            if (stack.hasCapability(CapabilityEnergy.ENERGY, null))
+                return  CapabilityProviderEnergy.getCap(stack).getEnergyStored();
+
+            return stack.getMaxDamage() - stack.getItemDamage();
+        }
+
+        /**
+         * Returns a Vec3i of the players position based on partial tick.
+         * Used for Render translation.
+         */
+        private static Vec3d getPlayerTranslate(EntityPlayer player, float partialTick) {
+            return new Vec3d(
+                    player.lastTickPosX + (player.posX - player.lastTickPosX) * partialTick,
+                    player.lastTickPosY + (player.posY - player.lastTickPosY) * partialTick,
+                    player.lastTickPosZ + (player.posZ - player.lastTickPosZ) * partialTick
+            );
+        }
+
+        /**
+         * Attempts to get the Silk Touch Drop item but if it fails it'll return the original
+         * non-silk touch ItemStack.
+         */
+        private static ItemStack getSilkDropIfPresent(World world, IBlockState state, EntityPlayer player) {
+            ItemStack itemStack = ItemStack.EMPTY;
+            if (state.getBlock().canSilkHarvest(world, BlockPos.ORIGIN, state, player))
+                itemStack = InventoryManipulation.getSilkTouchDrop(state);
+
+            if (itemStack.isEmpty())
+                itemStack = state.getBlock().getPickBlock(state, null, world, BlockPos.ORIGIN, player);
+
+            return itemStack;
+        }
+
+        private static void stateManagerPrepareBlend() {
+            GlStateManager.enableBlend();
+            GlStateManager.blendFunc(GL11.GL_CONSTANT_ALPHA, GL11.GL_ONE_MINUS_CONSTANT_ALPHA);
+        }
+
+        /**
+         * Prepares our render using base properties
+         * @param shouldShift if we should scale and shift the render to not cause issues with z-index
+         */
+        private static void stateManagerPrepare(Vec3d playerPos, BlockPos blockPos, boolean shouldShift, @Nullable Float shiftValue) {
+            GlStateManager.translate(blockPos.getX()-playerPos.x, blockPos.getY() - playerPos.y, blockPos.getZ() - playerPos.z);//Now move the render position to the coordinates we want to render at
+            // Rotate it because i'm not sure why but we need to
+            GlStateManager.rotate(-90.0F, 0.0F, 1.0F, 0.0F);
+
+            if( shouldShift && shiftValue != null ) {
```

Comment:
Why pass a flag and an Optional? If you already have the flag could you use float?

(or use Optional<Float> - whatever you prefer... (I presume Optional would be cleaner though, as one would atomatically trigger it, as soon as one passes the value. No need to handle the flag) )

Reply:
I actually don't know how you would do this with an optional as I've not used them much. I've replaced it with a nullable shiftValue instead of shouldShift

Reply:
Oh it's nice with an Optional...